resources/guidelines/testing/e2e-best-practises.md

```diff
@@ -77,3 +119,17 @@ cy.wait('@getData').then((xhr) => {
 });
 ```
 
+{% hint style="info" %} This [best practice](https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting) is also mentioned in Cypress' best practices as well. Actually, it can be considered as general best practice to avoid flakiness.
+{% endhint %}
+
+## Cypress' commands and their queue
+
+{% hint style="danger" %} Using vanilla JavaScript logic alongside cypress commands without further caution {% endhint %}
+
+{% hint style="success" %} If you need vanilla Javascript in your test, wrap it in a Cypress' `then` or build a custom command in order to get it queued. {% endhint %}
+
+Cypress commands are asynchronous and get queued for execution at a later time. During execution, subjects are yielded from one command to the next, and a lot of helpful Cypress code runs between each command to ensure everything is in order.
+
+This won't happen with Vanilla JS though, it will be executed immediately. In the worst case, this difference can cause timing issues. So always wrap your vannila JavaScript code into Cypress commands or `then` in order to make use of Cypress' command queue.
```

Comment:
```suggestion
This won't happen with Vanilla JS though, it will be executed immediately. In the worst case, this difference can cause timing issues. So always wrap your vanilla JavaScript code into Cypress commands or `then` in order to make use of Cypress' command queue.
```